nucypher/blockchain/eth/agents.py

```diff
@@ -288,6 +288,22 @@ def get_all_stakes(self, staker_address: str):
         for stake_index in range(stakes_length):
             yield self.get_substake_info(staker_address=staker_address, stake_index=stake_index)
 
+    @validate_checksum_address
+    def is_restaking(self, staker_address: str) -> bool:
+        staker_info = self.get_staker_info(staker_address)
+        restake_flag = bool(staker_info[3])
+        return restake_flag
+
+    @validate_checksum_address
+    def is_restake_locked(self, staker_address: str) -> bool:
```

Comment:
Do we need this and get_restaking_lock_status?

Reply:
There is already an indication of re-staking status in the CLI painting module - when running `nucypher status stakers`.   Perhaps it can also be included in `nucypher stake list`, I will take a look.

Reply:
I think that what @mswilkison means is that the code of `is_restake_locked()` and `get_restaking_lock_status()` is functionally identical, which is a consequence of merging our duplicated efforts. We just need one of them.

Reply:
De-duplicated.  :heavy_check_mark: